### PR TITLE
fix(lambda-python): asset files are generated inside the 'asset-input' folder

### PR DESCRIPTION
**Pull request by otaviomacedo**
Changes the asset structure from:

```
├── asset.993168cfa75b295eeea55bf603340284b9be46ebc079e4965f2c16f5470efda5
│   └── asset-input
│       ├── __init__.py
│       └── app.py
```

to:

```
├── asset.993168cfa75b295eeea55bf603340284b9be46ebc079e4965f2c16f5470efda5
│   ├── __init__.py
│   └── app.py
```

Fixes https://github.com/aws/aws-cdk/issues/18301.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-lambda-python/lib/bundling.ts [hunk 1]
```diff
@@ -86,8 +86,8 @@ export class Bundling implements CdkBundlingOptions {
     bundlingCommands.push(packaging.exportCommand ?? '');
     if (packaging.dependenciesFile) {
       bundlingCommands.push(`python -m pip install -r ${DependenciesFile.PIP} -t ${options.outputDir}`);
-    };
-    bundlingCommands.push(`cp -R ${options.inputDir} ${options.outputDir}`);
+    }
+    bundlingCommands.push(`cp -R ${options.inputDir}/* ${options.outputDir}`);
```
Comment (njlynch) [inline packages/@aws-cdk/aws-lambda-python/lib/bundling.ts:90]: I believe this syntax will drop hidden files, whereas `cp -R input/ output` will retain them.

Comment (otaviomacedo) [reply packages/@aws-cdk/aws-lambda-python/lib/bundling.ts:90]: Ok, let me test that.

Comment (otaviomacedo) [reply packages/@aws-cdk/aws-lambda-python/lib/bundling.ts]: Updated.

**Review by njlynch — APPROVED** [review]

**Comment by njlynch** [comment]
@mergifyio backport v2-main

**Pull request merged**